app/iot/service.py

```diff
@@ -39,11 +42,18 @@ def unregister_device(self, device_id: str) -> None:
     def get_device(self, device_id: str) -> Device:
         return self.devices[device_id]
 
-    def run_program(self, program: list[Message]) -> None:
+    async def run_program(self, program: list[Message]) -> None:
         print("=====RUNNING PROGRAM======")
-        for msg in program:
-            self.send_msg(msg)
+
+        messages = [
+            asyncio.create_task(self.send_msg(msg))
+            for msg in program
+        ]
+
+        gathered_massages = asyncio.gather(*messages)
```

Comment:
There is a typo in the variable name 'gathered_massages'. It should be 'gathered_messages'.